log: fix tags in traces

RaduBerinde opened the pull request:
The existing tracing code has a problem with log tags: when we trace through a
logging function like `Infof`, the tags show up in the events. When we trace
through an `Event` function, they do not.

Reworking the tracing code to provide a common function to log events, and
fixing this issue by always including log tags. We also fix `VEvent` functions
which were calling `Info/Infof` directly which would result in an incorrect
caller identification.

Longer term, we may want to be smarter: the tags that are part of the ancestor
context where the Span of EventLog was started should be included in the name of
the Span/EventLog (or set as span tags), and only the tags that were added to
the context later should appear in event messages. This is too complicated for
the amount of tags we currently have but we may revisit this in the future as
tag usage expands.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/8850)

<!-- Reviewable:end -->

andreimatei commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[util/log/structured.go, line 28 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNH8Cogxym9V2kMuyU:-KQNH8CpyXXZJJ_846ct:bwtjet5) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/structured.go#L28)):*
> ```Go
> )
> 
> // formatTags returns the tags as a bytes.Buffer.
> ```

"appends the tags to a buffer". And comment on the return val.
Or alternatively you can make this function allocate and return the buffer, although that's probably un-idiomatic.

---

*[util/log/structured.go, line 68 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNIh-reuMvh6tiyYTy:-KQNIh-reuMvh6tiyYTz:bjhxd9h) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/structured.go#L68)):*
> ```Go
> 	file, line, _ := caller.Lookup(depth + 1)
> 	msg := makeMessage(ctx, format, args)
> 	eventInternal(ctx, (s >= Severity_ERROR), false, "%s", msg)
> ```

please put the param name in `/*withTags */` and comment that `makeMessage` has already dealt with the tags.

---

*[util/log/trace.go, line 117 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNIz5LGZxLCGry8PVD:-KQNIz5MlKGZrgWBW8qw:bczw01m) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/trace.go#L117)):*
> ```Go
> // and logs the message to it. If neither is found, does nothing.
> func Event(ctx context.Context, msg string) {
> 	eventInternal(ctx, false, true, "%s", msg)
> ```

please comment the bool args. Or, since there's two of them in a row, maybe it's worth making them enums.

---

*[util/log/trace.go, line 157 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNJKT4wS_ZKhkpqqJb:-KQNJKT4wS_ZKhkpqqJc:b-a06yo3) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/trace.go#L157)):*
> ```Go
> // active trace or event log) or to the trace/event log alone, depending on
> // whether the specified verbosity level is active.
> func VEventf(level level, ctx context.Context, format string, args ...interface{}) {
> ```

can't this function go away, give your previous PR which dealt with the queues' use of the EventLog ?
Doesn't everybody just do `log.VTracef()` now? Maybe add a deprecation comment?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-:-KQNKPiiMoi2HXcyGSdy:bnfp4nl)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
log: fix tags in traces

The existing tracing code has a problem with log tags: when we trace through a
logging function like `Infof`, the tags show up in the events. When we trace
through an `Event` function, they do not.

Reworking the tracing code to provide a common function to log events, and
fixing this issue by always including log tags. We also fix `VEvent` functions
which were calling `Info/Infof` directly which would result in an incorrect
caller identification.

Longer term, we may want to be smarter: the tags that are part of the ancestor
context where the Span of EventLog was started should be included in the name of
the Span/EventLog (or set as span tags), and only the tags that were added to
the context later should appear in event messages. This is too complicated for
the amount of tags we currently have but we may revisit this in the future as
tag usage expands.

RaduBerinde commented:
TFTR!

---

Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[util/log/structured.go, line 28 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNH8Cogxym9V2kMuyU:-KQNWGPKPlIAoa_qdTMD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/structured.go#L28)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> "appends the tags to a buffer". And comment on the return val.
> Or alternatively you can make this function allocate and return the buffer, although that's probably un-idiomatic.
</details>
Done.

---

*[util/log/structured.go, line 68 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNIh-reuMvh6tiyYTy:-KQNWexTzQV_dMznvQTt:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/structured.go#L68)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> please put the param name in `/*withTags */` and comment that `makeMessage` has already dealt with the tags.
</details>
Done.

---

*[util/log/trace.go, line 117 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNIz5LGZxLCGry8PVD:-KQNX4DSVw60gNaEi1ua:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/trace.go#L117)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> please comment the bool args. Or, since there's two of them in a row, maybe it's worth making them enums.
</details>
Done.

---

*[util/log/trace.go, line 157 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-KQNJKT4wS_ZKhkpqqJb:-KQNX4nbcnGhoZdzP_8Z:bvyye81) ([raw file](https://github.com/cockroachdb/cockroach/blob/c36bf121b842072b1aca1b356f840ae87d370ec5/util/log/trace.go#L157)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> can't this function go away, give your previous PR which dealt with the queues' use of the EventLog ?
> Doesn't everybody just do `log.VTracef()` now? Maybe add a deprecation comment?
</details>
There was never any distinction between EventXX and TraceXX. I just renamed the TraceXX functions to EventXX and kept TraceXX with deprecation comments (I don't want to mess with all the callsites while we have to branches; they will go away after code-yellow). The reason was partly the conversation we had where I was incorrectly referring to events in spans as "traces".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8850#-:-KQNZX6QTERhCWjxRQ8R:b-814g8k)*
<!-- Sent from Reviewable.io -->
